### ISSUE TITLE
2.2.12 implementation doesn't meet task requirements

### ISSUE THREAD

**Issue by traditio** [lisztli/homework#2]
You misunderstood the task [2.2.12](https://github.com/lisztli/homework/blob/master/algorithms/ch2/2.2.12.py). It says to develop a **merge implementation** that uses max(M,N/M) space but you developed a **mergesort** implementation and it uses N space.

**Comment by aoeuidht** [comment]
You're right, I do allocate a N-sized list and use only a silce of that for M subsets. Will fix that soon.
